tsl/test/sql/data_node_bootstrap.sql

```diff
@@ -79,8 +79,46 @@ DELETE FROM _timescaledb_catalog.metadata WHERE key = 'dist_uuid';
 SELECT * FROM add_data_node('bootstrap_test', host => 'localhost',
                             database => 'bootstrap_test', bootstrap => false);
 
-SELECT * FROM delete_data_node('bootstrap_test');
+\set ON_ERROR_STOP 0
+-- Dropping the database with delete_data_node should not work in a
+-- transaction block since it is non-transactional.
+BEGIN;
+SELECT * FROM delete_data_node('bootstrap_test', drop_database => true);
```

Comment:
Might be good to have a case, when someone is still using the database and that prevents the drop database execution, yet not sure how to hold the connection here, maybe there is a similar way to do achieve this

Reply:
Yeah, I thought about that, but it probably requires an isolation test to spawn multiple concurrent processes.

I did verify that it will fail if the current session has an open connection, and that's why I added the code to purge the cache of open connections. Still, I think we are at that point just testing that DROP DATABASE fails the way it should, which I think we should be able to trust that it does.